File: LinkedLists/LeetCode/Merge_Two_Sorted_Lists.cs

namespace LinkedLists.LeetCode
{
    public class Merge_Two_Sorted_Lists
    {
        public ListNode MergeTwoLists(ListNode l1, ListNode l2)
        {
            // define two variable for head and movment variable
            var head = new ListNode(-1);
            var prev = head;

            //traversing in variable till l1 and l2 !=null
            while (l1 != null && l2 != null)
            {
                if (l1.val <= l2.val)
                {
                    prev.next = l1;
                    l1 = l1.next;
                }
                else
                {
                    prev.next = l2;
                    l2 = l2.next;
                }
                prev = prev.next;
            }
            prev.next = l1 != null ? l1 : l2;

            //if l1 or l2 length bigger than each other , involve
            return head.next;
        }

    }
}

File: LinkedLists/LeetCode/Merge_Two_Sorted_Lists.go
package LinkedList

func main() {

}

type ListNode struct {
	Val  int
	Next *ListNode
}

func mergeTwoLists(l1 *ListNode, l2 *ListNode) *ListNode {
	if l1 == nil {
		return l2
	}
	if l2 == nil {
		return l1
	}

	main := &ListNode{Val: -1, Next: nil}
	head := main

	for l1 != nil && l2 != nil {
		if l1.Val < l2.Val {
			head.Next = l1
			l1 = l1.Next
		} else {
			head.Next = l2
			l2 = l2.Next
		}
		head = head.Next

	}

	if l1 == nil {
		head.Next = l2
	} else {
		head.Next = l1
	}

	return main.Next

}
